provider/offline.go

```diff
@@ -372,7 +372,7 @@ func (rec *ResourceRecord) SetEntity(entity interface{}) error {
 	case string:
 		rec.Entity = typedEntity
 	default:
-		return fmt.Errorf("entity must be a string; received %T", entity)
+		return fferr.NewInvalidArgument(fmt.Errorf("entity must be a string; received %T", entity))
```

Comment:
Internal or type error?

Reply:
I actually think it's fine to use invalid arg given the user is the one who's responsible for this error.

Reply:
thats fine